utils/s2n_safety.h

```diff
@@ -163,6 +163,16 @@
  */
 #define POSTCONDITION_POSIX( condition )            __S2N_ENSURE_CONDITION((condition), BAIL_POSIX(S2N_ERR_POSTCONDITION_VIOLATION))
 
+/**
+ * Ensures the `condition` is `true`, otherwise the function will `BAIL_PTR` with a `S2N_ERR_PRECONDITION_VIOLATION` error
+ */
+#define PRECONDITION_PTR( condition )             __S2N_ENSURE_CONDITION((condition), BAIL_PTR(S2N_ERR_PRECONDITION_VIOLATION))
+
+/**
+ * Ensures the `condition` is `true`, otherwise the function will `BAIL_PTR` with a `S2N_ERR_POSTCONDITION_VIOLATION` error
+ */
+#define POSTCONDITION_PTR( condition )            __S2N_ENSURE_CONDITION((condition), BAIL_PTR(S2N_ERR_POSTCONDITION_VIOLATION))
+
```

Comment:
Not anymore here, but I guess we should keep them around since we're not removing the other pre- and post-conditions macros, right? @camshaft

Reply:
It depends on if we want to encourage the use of pre/post condition macros. My preference would be no because it doesn't provide very good error messages. My preference would be to use the `_validate` convention long term.